### PR TITLE
Add tests for valid and invalid datasource names

### PR DESCRIPTION
**Pull request by suneet-s**

### CODE REVIEW

File: server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java [hunk 1]
```diff
@@ -54,10 +56,20 @@
...
 public class DataSchemaTest
 {
+
+  private static final String VALID_DATASOURCE_CHARS_NAME = "alpha123..*~!@#&%^&*()-+ Россия\\ 한국 中国!";
+  private static final String VALID_DATASOURCE_CHARS_FOR_SERIALIZER = "alpha123..*~!@#&%^&*()-+ Россия\\\\ 한국 中国!";
+
+  static {
+    // testTransformSpec needs NullHandling initialized.
+    NullHandling.initializeForTests();
+  }
```
Comment (ccaominh) [inline server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java:71]: The test class can extend `NullHandlingTest` instead.

File: server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java [hunk 2]
```diff
@@ -54,10 +56,20 @@
...
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 
 public class DataSchemaTest
 {
+
+  private static final String VALID_DATASOURCE_CHARS_NAME = "alpha123..*~!@#&%^&*()-+ Россия\\ 한국 中国!";
+  private static final String VALID_DATASOURCE_CHARS_FOR_SERIALIZER = "alpha123..*~!@#&%^&*()-+ Россия\\\\ 한국 中国!";
```
Comment (ccaominh) [inline server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java:66]: Suggest adding a helper method that escapes `VALID_DATASOURCE_CHARS_NAME`. The helper method can be used here as well as in `testSerializeWithInvalidDataSourceName()`

File: server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java (hunk 1 above)
Comment (suneet-s) [reply server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java:71]: Done

File: server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java (hunk 2 above)
Comment (suneet-s) [reply server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java:66]: 👍

File: server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java [hunk 3]
```diff
@@ -54,10 +57,14 @@
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 
-public class DataSchemaTest
+public class DataSchemaTest extends NullHandlingTest
```
Comment (clintropolis) [inline server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java:63]: nit: it's actually `InitializedNullHandlingTest` that you want, but the one you've extended also initializes null handling

**Pull request merged**